### PR TITLE
Use smart-open for opening historical files

### PR DESCRIPTION
**Pull request by petedmarsh**
>smart_open is a drop-in replacement for Python's built-in open():
it can
>do anything open can (100% compatible, falls back to native open
>wherever possible), plus lots of nifty extra stuff on top.

The main bit of "nifty extra stuff" it is being used for here is for transparent decompression of compressed files.

Previously in docs/performance.md smart-open was mention and a monkey patch was provided:

    with patch("builtins.open", smart_open.open):
        framework.add_strategy(strategy)
        framework.run()

Anyone who was using this monkey patch will not have their workflow distrupted by this change (as I have essentially hard-coded in what was being patched in).

At the time of writing the latest release version of smart-open is 6.3.0.

https://github.com/RaRe-Technologies/smart_open

### CODE REVIEW

**Comment by petedmarsh** [comment]
Fixes: https://github.com/betcode-org/flumine/issues/688

File: docs/performance.md [hunk 1]
```diff
@@ -40,19 +40,9 @@ logger.setLevel(logging.CRITICAL)
 
 ### File location
 
-This might sound obvious but having the market files stored locally on your machine will allow much quicker processing. A common pattern is to use s3 to store all market files but a local cache for common markets processed.
```
Comment (liampauling) [inline docs/performance.md:43]: Why remove this hint?

Comment (petedmarsh) [reply docs/performance.md:43]: It hasn't been removed, just moved :)

Comment (liampauling) [reply docs/performance.md:43]: 🤦

**Review by liampauling — APPROVED** [review]

**Pull request merged**